### PR TITLE
[docs] Enhance container selection in docker dump

### PR DESCRIPTION
**Pull request by openscript**
The problem with the previous query was, that it sometimes selected multiple containers, which make the command respond with a hard to understand error message.

With this regular expression only the container is selected with the exact name.

### CODE REVIEW

File: docs/content/doc/usage/backup-and-restore.en-us.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ The command has to be executed with the `RUN_USER = <OS_USERNAME>` specified in
 Example:
 
 ```none
-docker exec -u <OS_USERNAME> -it -w <--tempdir> $(docker ps -qf "name=<NAME_OF_DOCKER_CONTAINER>") bash -c '/app/gitea/gitea dump -c </path/to/app.ini>'
+docker exec -u <OS_USERNAME> -it -w <--tempdir> $(docker ps -qf name=^/<NAME_OF_DOCKER_CONTAINER>$) bash -c '/app/gitea/gitea dump -c </path/to/app.ini>'
```
Comment (6543) [inline docs/content/doc/usage/backup-and-restore.en-us.md]:
```suggestion
docker exec -u <OS_USERNAME> -it -w <--tempdir> $(docker ps -qf name=^<NAME_OF_DOCKER_CONTAINER>$) bash -c '/app/gitea/gitea dump -c </path/to/app.ini>'
```
do you mean this?

**Review by 6543 — APPROVED** [review]

File: docs/content/doc/usage/backup-and-restore.en-us.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ The command has to be executed with the `RUN_USER = <OS_USERNAME>` specified in
 Example:
 
 ```none
-docker exec -u <OS_USERNAME> -it -w <--tempdir> $(docker ps -qf "name=<NAME_OF_DOCKER_CONTAINER>") bash -c '/app/gitea/gitea dump -c </path/to/app.ini>'
+docker exec -u <OS_USERNAME> -it -w <--tempdir> $(docker ps -qf name=^<NAME_OF_DOCKER_CONTAINER>$) bash -c '/app/gitea/gitea dump -c </path/to/app.ini>'
```
Comment (zeripath) [inline docs/content/doc/usage/backup-and-restore.en-us.md]: are you sure that name argument does not still need to be in quotes?

**Comment by lunny** [comment]
@openscript Any update?

**Comment by openscript** [comment]
@lunny I'll check it today and give some feedback :)

**Comment by openscript** [comment]
@lunny I can verify that it works without the quotes. This is I usually do my backups:

 1. Create a new backup <br> `docker exec -u git -it -w /tmp $(docker ps -qf name=^/gitea$) bash -c '/app/gitea/gitea dump -c /data/gitea/conf/app.ini -f gitea-backup.zip'`
 1. Copy backup <br> `docker cp $(docker ps -qf name=^/gitea$):/tmp/gitea-backup.zip .`
 1. Rename backup file <br> `mv gitea-backup.zip gitea-backup-$(date "+%Y%m%d-%H%M%S").zip`

Maybe this section in the documentation can be enhanced further. When no file name for the dump is specified, a timestamp will be used. So you need to go into the container to find the actual file name.

File: docs/content/doc/usage/backup-and-restore.en-us.md (hunk 2 above)
Comment (openscript) [reply docs/content/doc/usage/backup-and-restore.en-us.md]: Yes I'm sure.

Comment (zeripath) [reply docs/content/doc/usage/backup-and-restore.en-us.md]: This is shell dependent. If you are running the zsh the ^ and $ will be picked up an interpreted by the shell.

Comment (openscript) [reply docs/content/doc/usage/backup-and-restore.en-us.md]: We use this with `bash`, but I agree.. this is probably true.

Comment (zeripath) [reply docs/content/doc/usage/backup-and-restore.en-us.md]: So could you stick it in quotes as suggested then I will be happy to approve.

**Comment by zeripath** [comment]
Without quotes it is shell dependent. Bash works fine but zsh definitely will not.

File: docs/content/doc/usage/backup-and-restore.en-us.md (hunk 2 above)
Comment (delvh) [reply docs/content/doc/usage/backup-and-restore.en-us.md]:
```suggestion
docker exec -u <OS_USERNAME> -it -w <--tempdir> $(docker ps -qf 'name=^<NAME_OF_DOCKER_CONTAINER>$') bash -c '/app/gitea/gitea dump -c </path/to/app.ini>'
```
will always work.

**Comment by wxiaoguang** [comment]
I think @delvh 's suggestion is pretty good. So, approve and move on?

**Review by 6543 — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**